### PR TITLE
Fix Swift 5 errors and warnings.

### PR DESCRIPTION
**Pull request by BjornRuud**
Some changes were needed to compile with Swift 5 in Xcode 10.2.

### CODE REVIEW

**Comment by gaetanzanella** [comment]
I think you should update the .podspec accordingly:
```
spec.swift_version = "4.2"
spec.swift_versions = ['4.2', '5.0' ] # cocoapods 1.7
```

**Comment by parrots** [comment]
There's a previous PR for Swift 5 work. Might be good to consolidate into that one (including the Podspec changes?) https://github.com/ra1028/DifferenceKit/pull/55

File: Sources/Extensions/UIKitExtension.swift [hunk 1]
```diff
@@ -23,12 +23,12 @@ public extension UITableView {
         ) {
         reload(
             using: stagedChangeset,
-            deleteSectionsAnimation: animation,
-            insertSectionsAnimation: animation,
-            reloadSectionsAnimation: animation,
-            deleteRowsAnimation: animation,
-            insertRowsAnimation: animation,
-            reloadRowsAnimation: animation,
+            deleteSectionsAnimation: animation(),
```
Comment (Kaspik) [inline Sources/Extensions/UIKitExtension.swift]: Even tho podspec changes say supported Swift 4.2 - this is not supported in Swift 4.2. Actually this PR is more like migration than just updates. 😄

Comment (BjornRuud) [reply Sources/Extensions/UIKitExtension.swift]: That's strange, the project has been set to use Swift 4.2 and compiles just fine here. Or do you mean something else besides compilation is the problem?

Comment (BjornRuud) [reply Sources/Extensions/UIKitExtension.swift]: Those changes are necessary because autoclosure forwarding behaviour has changed.

**Comment by BjornRuud** [comment]
Duplicate of #55

**Comment by ra1028** [comment]
Thank you, you also did a good job 👍

**Pull request closed without merging**